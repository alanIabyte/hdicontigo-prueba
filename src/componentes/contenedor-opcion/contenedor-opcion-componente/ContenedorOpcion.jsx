import React from "react";
import PropTypes from "prop-types";
// import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
// import { faChevronRight } from "@fortawesome/free-solid-svg-icons";
import {
  EnvolvedorContenedorOpcion,
  ContenedorElementos,
  Titulo,
  Subtitulo,
  Icono,
  // Indicador,
} from "./ContenedorOpcion.styled";

class ContenedorOpcion extends React.Component {
  render() {
    const { titulo, subtitulo, url, icono } = this.props;
    return (
      <EnvolvedorContenedorOpcion href={url}>
        <Icono> {icono} </Icono>
        <ContenedorElementos>
          <Titulo> {titulo} </Titulo>
          <Subtitulo> {subtitulo} </Subtitulo>
        </ContenedorElementos>
        {/* <Indicador>
          <FontAwesomeIcon icon={faChevronRight} />
        </Indicador> */}
      </EnvolvedorContenedorOpcion>
    );
  }
}

ContenedorOpcion.propTypes = {
  titulo: PropTypes.string,
  subtitulo: PropTypes.string,
  url: PropTypes.string,
  icono: PropTypes.node,
};

ContenedorOpcion.defaultProps = {
  titulo: "",
  subtitulo: "",
  url: "#",
  icono: null,
};

export default ContenedorOpcion;
