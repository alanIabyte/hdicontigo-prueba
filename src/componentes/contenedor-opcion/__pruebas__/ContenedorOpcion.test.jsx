import React from "react";
import renderer from "react-test-renderer";
import ContenedorOpcion from "../contenedor-opcion-componente/ContenedorOpcion";
import ContenedorOpcionSimulacion from "../__simulaciones__/ContenedorOpcion.simulacion";

describe("ContenedorOpcion", () => {
  it("se despliega correctamente", () => {
    const { titulo, subtitulo, url, icono } = ContenedorOpcionSimulacion;
    const tree = renderer
      .create(
        <ContenedorOpcion
          titulo={titulo}
          subtitulo={subtitulo}
          url={url}
          icono={icono}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
