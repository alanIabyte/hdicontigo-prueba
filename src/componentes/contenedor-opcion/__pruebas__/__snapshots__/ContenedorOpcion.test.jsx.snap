// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContenedorOpcion se despliega correctamente 1`] = `
<a
  className="sc-aXZVg hzQrdN"
  href="http://test"
>
  <div
    className="sc-iGgWBj dGyqTJ"
  >
     
    http://test
     
  </div>
  <div
    className="sc-gEvEer onmco"
  >
    <div
      className="sc-eqUAAy kHRksZ"
    >
       
      titulo
       
    </div>
    <div
      className="sc-fqkvVR ehrpfY"
    >
       
      subtitulo
       
    </div>
  </div>
</a>
`;
